lib/pure/net.nim

```diff
@@ -611,6 +606,9 @@ when defineSsl:
     when not defined(openssl10) and not defined(libressl):
       let sslVersion = getOpenSSLVersion()
       if sslVersion >= 0x010101000 and not sslVersion == 0x020000000:
+        # XXX always false!
+        # XXX however, setting non-1.3 ciphers with ciphersuites will
+        # XXX cause an error, cipherList needs to be split into 1.3 and non-1.3
```

Comment:
It does? According to the [man page](https://www.openssl.org/docs/manmaster/man3/SSL_CTX_set_ciphersuites.html):

> Items that are not recognized, because the corresponding ciphers are not compiled in or because they are mistyped, are simply ignored. Failure is only flagged if no ciphers could be collected at all.

Am I missing anything here?

Reply:
Ah, I see, if you have a cipher list without 1.3 ciphers then it would error out.

I really hate how openssl decides that it needs two functions for this.

Reply:
found it when testing this, trying to specify only a PSK cipher